tests/test_push_deduced_required.py

```diff
@@ -0,0 +1,62 @@
+"""Copyright 2019 Equinor ASA and The Netherlands Organisation for
+Applied Scientific Research TNO.
+
+Licensed under the MIT license.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the conditions stated in the LICENSE file in the project root for
+details.
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+"""
+
+import unittest
+
+import configsuite
+
+from . import data
+
+
+class TestPush(unittest.TestCase):
+    def assertEqualSnapshots(self, first, second):
+        self.assertEqual(first.heroes, second.heroes)
+        self.assertEqual(sorted(first.villains), sorted(second.villains))
+
+    def test_push_deduced_required(self):
```

Comment:
There are already tests on `push` in `test_layers.py`. Should we move this test over in that file? Or potentially move all tests on push into this test?

Reply:
Might make sense to move it, my main problem is that this isn´t a failing test, although I first thought it would be.

Reply:
So just to clarify for me a bit, because containers can no longer be required, must the programmer add a custom validation to make sure they are specified?

Reply:
Yeah, that should be fixed first! In that respect it might make sense to move it to `test_depreation_warning.py`. You should be able to build on one of the tests there to verify that `deduce_required` also is active after a push?

Reply:
Will push an actual failing test in a sec